controller/users_blackbox_test.go

```diff
@@ -984,3 +994,71 @@ func createDummyUserProfileResponse(updatedBio, updatedImageURL, updatedURL *str
 	return profile
 
 }
+
+func (s *TestUsersSuite) TestCreateUserAsServiceAccountOK() {
+
+	// given
+	user := testsupport.TestUser
+	identity := testsupport.TestIdentity
+	identity.User = user
+	identity.ProviderType = "KC"
+
+	user.ContextInformation = map[string]interface{}{
+		"last_visited": "yesterday",
+		"space":        "3d6dab8d-f204-42e8-ab29-cdb1c93130ad",
+		"rate":         100.00,
+		"count":        3,
+	}
+	secureService, secureController := s.SecuredServiceAccountController(testsupport.TestOnlineRegistrationAppIdentity)
+
+	// when
+	createUserPayload := createCreateUsersAsServiceAccountPayload(&user.Email, &user.FullName, &user.Bio, &user.ImageURL, &user.URL, &user.Company, &identity.Username, &identity.RegistrationCompleted, user.ContextInformation, user.Cluster)
+	_, appUser := test.CreateUsersOK(s.T(), secureService.Context, secureService, secureController, createUserPayload)
+	assertCreatedUser(s.T(), appUser.Data, user, identity)
+}
+
+func (s *TestUsersSuite) TestCreateUserAsServiceAccountUnauthorized() {
```

Comment:
Good. You test if there is the expected SA or if there is a SA token but with unexpected service name.
But let's also test:
- if there is no token at all.
- if the token is not a service account token but a regular OSIO user token.

Reply:
Thanks, added ab4347d